libs/langchain/langchain/prompts/few_shot.py

```diff
@@ -1,24 +1,25 @@
 """Prompt template that contains few shot examples."""
-from typing import Any, Dict, List, Optional
+from __future__ import annotations
 
-from pydantic import Extra, root_validator
+from typing import Any, Dict, List, Optional, Union
+
+from pydantic import BaseModel, Extra, root_validator
 
 from langchain.prompts.base import (
     DEFAULT_FORMATTER_MAPPING,
     StringPromptTemplate,
     check_valid_template,
 )
+from langchain.prompts.chat import BaseChatPromptTemplate, BaseMessagePromptTemplate
 from langchain.prompts.example_selector.base import BaseExampleSelector
 from langchain.prompts.prompt import PromptTemplate
+from langchain.schema.messages import BaseMessage, get_buffer_string
+from langchain.schema.prompt_template import BasePromptTemplate
 
 
-class FewShotPromptTemplate(StringPromptTemplate):
+class _FewShotPromptTemplateMixin(BaseModel):
```

Comment:
does mixin need to be BaseModel (seems like everything that uses it already is)

Reply:
It was giving some weird metaclass resolution things without but I'll double check that still is the case